srml/grandpa/src/lib.rs

```diff
@@ -376,3 +378,53 @@ impl<T: Trait> finality_tracker::OnFinalizationStalled<T::BlockNumber> for Modul
 		<Stalled<T>>::put((further_wait, median));
 	}
 }
+
+// TODO [slashing]: Integrate this.
+/// A grandpa equivocation offence report.
+#[allow(dead_code)]
+struct GrandpaEquivocationOffence<FullIdentification> {
+	/// A round in which the incident happened.
+	round: u64,
```

Comment:
round resets to 0 every set. This should contain the Set ID too.

Reply:
The offences module will use `(SessionIndex, Round)` as a key, do you think that's sufficient?

Reply:
Yeah, although I don't know how you'll go from set-id to session index